src/htmldrawer.js

```diff
@@ -0,0 +1,239 @@
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+(function( $ ){
+
+/**
+ * @class HTMLDrawer
+ * @memberof OpenSeadragon
+ * @classdesc HTML-based implementation of DrawerBase for an {@link OpenSeadragon.Viewer}.
+ * @param {Object} options - Options for this Drawer.
+ * @param {OpenSeadragon.Viewer} options.viewer - The Viewer that owns this Drawer.
+ * @param {OpenSeadragon.Viewport} options.viewport - Reference to Viewer viewport.
+ * @param {Element} options.element - Parent element.
+ * @param {Number} [options.debugGridColor] - See debugGridColor in {@link OpenSeadragon.Options} for details.
+ */
+
+class HTMLDrawer extends $.DrawerBase{
+    constructor(options){
+        super(options);
+    }
+
+    /**
+     * @returns {Boolean} always true
+     */
+    static isSupported(){
+        return true;
+    }
+
+    getType(){
+        return 'html';
+    }
+
+    /**
+     * @returns {Boolean} Whether this drawer requires enforcing minimum tile overlap to avoid showing seams.
+     */
+    minimumOverlapRequired() {
+        return true;
+     }
+
+    /**
+     * create the HTML element (e.g. canvas, div) that the image will be drawn into
+     * @returns {Element} the div to draw into
+     */
+    _createDrawingElement(){
+        let canvas = $.makeNeutralElement("div");
+        return canvas;
+    }
+
+    /**
+     * Draws the TiledImages
+     */
+    draw(tiledImages) {
+        var _this = this;
+        this._prepareNewFrame(); // prepare to draw a new frame
+        tiledImages.forEach(function(tiledImage){
+            if (tiledImage.opacity !== 0 || tiledImage._preload) {
+                _this._drawTiles(tiledImage);
+            }
+        });
+
+    }
+
+    /**
+     * @returns {Boolean} False - rotation is not supported.
+     */
+    canRotate() {
+        return false;
+    }
+
+    /**
+     * Destroy the drawer (unload current loaded tiles)
+     */
+    destroy() {
+        //force unloading of current canvas (1x1 will be gc later, trick not necessarily needed)
+        this.canvas.innerHTML = "";
+    }
+
+    /**
+     * Turns image smoothing on or off for this viewer. Note: Ignored by HTML Drawer
+     *
+     * @function
+     * @param {Boolean} [imageSmoothingEnabled] - Whether or not the image is
+     * drawn smoothly on the canvas; see imageSmoothingEnabled in
+     * {@link OpenSeadragon.Options} for more explanation.
+     */
+    setImageSmoothingEnabled(){
+        // noop - HTML Drawer does not deal with this property
+        $.console.warn('HTMLDrawer.setImageSmoothingEnabled does not have an effect.');
+    }
+
+    /**
+     * @private
+     * Clears the Drawer so it's ready to draw another frame.
+     *
+     */
+    _prepareNewFrame() {
+        this.canvas.innerHTML = "";
```

Comment:
So every frame we delete all of the images in the div and recreate them? Seems highly inefficient. I guess that's what we've always done, though. Pretty strange.

Fixing that is outside the scope of this PR... Just commenting since I noticed it.

Reply:
Agreed... didn't want to fix what wasn't broken though. I do wonder how much the HTML drawer is in use "in the wild" these days.

Reply:
I know it is still used in some scenarios where the canvas drawer has problems. Perhaps it won't be used as much once the WebGL drawer becomes available. It's nice having it as a fallback in the meantime, but yeah, we don't need to be putting a lot of energy into it.